### PR TITLE
Fit isar

### PR DESCRIPTION
**Pull request by TigranH13**

### CODE REVIEW

File: lib/core/migration/miagrate_hive_to_isar.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  @override
+  String get id => 'migrationn';
+
+  Future<Isar> _openDb() async {
+    if (Isar.instanceNames.isEmpty) {
+      final dir = await getApplicationDocumentsDirectory();
+      return await Isar.open([TaskSchema],
+          inspector: true, directory: dir.path);
```
Comment (SevakBaghryan) [inline lib/core/migration/miagrate_hive_to_isar.dart:22]: end comma

File: lib/data/repository/todos_impl.dart
Comment (SevakBaghryan) [inline lib/data/repository/todos_impl.dart:1]: don't remain old files

File: lib/data/repository/todos_impl_wtih_isr.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  TodosRepositoryImplWithIsr() {
+    db = _openDb();
+  }
+
+  Future<Isar> _openDb() async {
+    if (Isar.instanceNames.isEmpty) {
+      final dir = await getApplicationDocumentsDirectory();
+      return await Isar.open([TaskSchema],
+          inspector: true, directory: dir.path);
```
Comment (SevakBaghryan) [inline lib/data/repository/todos_impl_wtih_isr.dart:18]: end comma

File: lib/locator.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'package:get_it/get_it.dart';
+import 'package:to_do_app/application/bloc/todo_bloc.dart';
+import 'package:to_do_app/data/repository/todos_impl_wtih_isr.dart';
+import 'package:to_do_app/domain/repository/todos.dart';
+
+final sl = GetIt.instance;
+
+Future<void> setup() async {
+  print('anuma');
```
Comment (SevakBaghryan) [inline lib/locator.dart:9]: remove print statements

File: lib/presentantion/screens/home_screen.dart [hunk 4]
```diff
@@ -16,9 +17,9 @@ class HomeScreen extends StatelessWidget {
         builder: (cn, state) {
           if (state is LoadedState) {
             List tasks = state.toDo;
-            tasks.sort(
-              ((a, b) => a.id.compareTo(b.id)),
-            );
+            // tasks.sort(
+            //   ((a, b) => a.id.compareTo(b.id)),
+            // );
```
Comment (SevakBaghryan) [inline lib/presentantion/screens/home_screen.dart:22]: remove comments

File: lib/presentantion/screens/home_screen.dart [hunk 5]
```diff
@@ -33,15 +34,15 @@ class HomeScreen extends StatelessWidget {
                           Text(task.title),
                           ElevatedButton(
                             onPressed: () {
-                              context.read<TodoBloc>().add(
-                                    DeleteTaskEvent(id: task.id),
-                                  );
+                              context
+                                  .read<TodoBloc>()
+                                  .add(DeleteTaskEvent(task: task));
```
Comment (SevakBaghryan) [inline lib/presentantion/screens/home_screen.dart:39]: end comma

File: lib/presentantion/screens/home_screen.dart [hunk 6]
```diff
@@ -33,15 +34,15 @@ class HomeScreen extends StatelessWidget {
...
                             },
                             child: const Text('delete'),
                           ),
                           ElevatedButton(
                               onPressed: () {
-                                context.router.push(EditRoute(toDo: task));
+                                context.router.push(EditRoute(task: task));
                               },
                               child: const Text('edit'))
```
Comment (SevakBaghryan) [inline lib/presentantion/screens/home_screen.dart:47]: end comma

File: lib/application/bloc/todo_bloc.dart [hunk 7]
```diff
@@ -66,8 +69,8 @@ class TodoBloc extends Bloc<TodoEvent, TodoState> {
     return on<SaveTaskEvent>(
       (event, emit) async {
         emit(const LoadingState());
-        await TodosRepositoryImpl().saveTask(task: event.task);
-        final todos = await todosRepositoryImpl.loadTasks();
+        await todosRepositoryImplWithIsr.saveTask(task: event.task);
+        final todos = await todosRepositoryImplWithIsr.loadTasks();
 
         emit(
           LoadedState(toDo: todos),
```
[the remark this answers is not in the document]
Comment (TigranHayrapetyan13) [reply lib/application/bloc/todo_bloc.dart:76]: it was joke

**Pull request merged**